### ISSUE TITLE
Accessibility bug when selecting items with space from the filter category

### ISSUE THREAD

**Issue by AnaMikkonen** [microsoft-search/pnp-modern-search#3811]
**Version used**
4.11.1

**Describe the bug**
We have found one major accessibility bug from the Modern Search V4 filters webpart.
It's possible to travel between filter categories by using tab key and between individual items in each category by using arrow keys. This is just like it should be! Problem occurs when I'm selecting an item by pressing a space key. The item is selected but the focus will be lost from the current location. This is a major accessibility issue which is pointed out on accessibility tests on few projects.
 

**To Reproduce**
Detailed steps to reproduce the behavior:
1. Configure search page you like with search box, results and filters webparts.  There should be at least one filter defined.
2. Execute a search with keyword that returns filterable results.
3. Press tab when cursor is on the search box until you hit the filter category title.
4. Travel to category item you want to select with (up and down) arrow keys. 
5.  Select item by pressing a space bar.
6.  Now the item is selected but you cannot travel to next item with arrow keys because the focus is "somewhere else".

**Expected behavior**
Focus should remain on the item when selected with space bar.

**Desktop (please complete the following information):**
 - Browser: Chrome, Version 126.0.6478.62 (Official Build) (64-bit)

**Comment by kasperbolarsen** [comment]
We would welcome a PR that would improve keyboard accessibility, but please be aware that this is not a priority for the project as we are not required to meet these accessibility requirements
